staging/src/k8s.io/apiserver/pkg/server/lifecycle_signals.go

```diff
@@ -188,3 +204,11 @@ func (e *namedChannelWrapper) Signaled() <-chan struct{} {
 func (e *namedChannelWrapper) Name() string {
 	return e.name
 }
+
+func (e *namedChannelWrapper) SignaledAt() *time.Time {
+	value := e.signaledAt.Load()
```

Comment:
It would be a bit surprising if SignaledAt returns nil if the signal hasn't fire yet but the converse is not also true. If you ran something like:

```go
<-Signaled()
ts := SignaledAt()
```

then you could end up with `ts` being nil, couldn't you?